tests/model/test_model.py

```diff
@@ -656,71 +656,110 @@ def test_append_message(self, mocker, user_profile, response,
+                2: {
+                    'id': 2,
+                    'content': 'old content',
+                    'subject': 'old subject'
+                }},
+            'edited_messages': {1}
         }),
         ({
             'rendered_content': '<p>new content</p>',
             'subject': 'new subject',
             'message_ids': [1],
         }, 2, {
-            'subject': 'new subject',
-            'content': '<p>new content</p>',
+            'messages': {
+                1: {
+                    'id': 1,
+                    'content': '<p>new content</p>',
+                    'subject': 'new subject'
+                },
+                2: {
+                    'id': 2,
+                    'content': 'old content',
+                    'subject': 'old subject'
+                }},
+            'edited_messages': {1}
         }),
         ({}, 0, {
-            'subject': 'old subject',
-            'content': 'old content',
+            'messages': {
+                1: {
+                    'id': 1,
+                    'content': 'old content',
+                    'subject': 'old subject'
+                },
+                2: {
+                    'id': 2,
+                    'content': 'old content',
+                    'subject': 'old subject'
+                }},
+            'edited_messages': {1}
         }),
     ])
-    def test_update_message(self, mocker, model, response, expected_message,
+    def test_update_message(self, mocker, model, response, new_index,
                             update_call_count, msg_id=1):
         model.index = {
             'messages': {
-                msg_id: {
-                    'id': msg_id,
+                message_id: {
+                    'id': message_id,
                     'content': 'old content',
                     'subject': 'old subject',
-                },
+                } for message_id in [1, 2]
             },
             'edited_messages': set()
         }
-        model.msg_list = mocker.Mock()
-        mock_msg = mocker.Mock()
-        model.msg_list.log = [mock_msg]
-        mock_msg.original_widget.message = model.index['messages'][msg_id]
-        mocker.patch('zulipterminal.model.create_msg_box_list',
-                     return_value=[mock_msg])
         mocker.patch('zulipterminal.model.Model.update_rendered_view')
         response['message_id'] = msg_id
 
         model.update_message(response)
 
-        assert model.index['messages'][msg_id]['content'] == \
-            expected_message['content']
-        assert model.index['messages'][msg_id]['subject'] == \
-            expected_message['subject']
-        assert model.msg_list.log[0] == mock_msg
-        # Ensure the index shows the message has been updated (edited)
-        assert model.index['edited_messages'] == {msg_id}
+        assert model.index == new_index
         assert model.update_rendered_view.call_count == update_call_count
 
-        # TEST FOR FALSE CASES
```

Comment:
This should be in a separate test, rather than removed completely? eg. `test_update_message_where_message_not_in_index` ?

Reply:
The last commit checks for this case.